webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/git/ZipUtils.scala

```diff
@@ -59,7 +59,14 @@ import zio.syntax.*
 
 object ZipUtils {
 
-  final case class Zippable(path: String, useContent: Option[(InputStream => IOResult[Any]) => IOResult[Any]])
+  // we need to ensure that path is never null, which can happen because we interact with Java
+  final case class Zippable(path: String, useContent: Option[(InputStream => IOResult[Any]) => IOResult[Any]]) {
+    if (null == path) {
+      throw new IllegalArgumentException(
+        s"A zippable can not be created with an 'null' path (useContent ${if (useContent.isEmpty) "is empty" else "is defined"})"
+      )
+    }
+  }
```

Comment:
Because there is a factory `Zippable#make`, is could be more idiomatic to put the null check in the constructor instead (and make the constructor private).
Also if we return an `Option[Zippable]`, we could also handle the error at the place where the zippable object needs to be used

Reply:
I played with that way, but it was meh : 
- what is idiomatic is to have the general contract that things are not null. Almost nowhere, we assume that a parameter can be null, because we do scala
- so the general usage is standard scala `case class`, with public constructor and caller not doing wrong things. 
- since we do have prior art of things passing null at that point, we add a guard to show that it's a developper error, and that it shall not be done like that. 
- the correct ansmwer is to have `Nullable` as a separed aspect of types and scala API forbidding it, but it's an ecosystem matter. 

For the factory part, I'm not sure why I reverted the "`Option` return statement. I think I was all-in in the mood of "if you return null, you have to deal with that, no defensive code that complexify everything everywhere"

Reply:
Ah, yes: the problem is that the factory just doesn't know what to do with a null. It's already too late in the code to handle it correctly, so it's really a dev error, and so an exception is the correct way to signal that the dev did something very bad: 
![image](https://github.com/user-attachments/assets/0ba13f2c-6c9f-45e6-95ab-f3b6d5cccb46)